src/commonMain/kotlin/fr/acinq/bitcoin/Script.kt

```diff
@@ -785,10 +785,11 @@ public object Script {
                         }
                     }
 
-                    head == OP_NOTIF && conditions.any { !it } -> {
+                    op == OP_NOTIF && conditions.any { !it } -> {
                         conditions.add(0, true)
                     }
-                    head == OP_NOTIF -> {
+                    op == OP_NOTIF && stack.isEmpty() -> throw RuntimeException("Invalid OP_IF construction")
```

Comment:
nit:
```suggestion
                    op == OP_NOTIF && stack.isEmpty() -> throw RuntimeException("Invalid OP_NOTIF construction")
```

Also `OP_ELSE` and `OP_ENDIF` have the same copy-paste issue.

Reply:
Bitcoin core uses the same error message for IF/NOTIF/... errors but you're right it's better to be more specific